Update package name references

pascalrioux opened the pull request:
- Update package name
- Rename dependencies to use the new Charcoal structure

pascalrioux committed:
Update composer.json

mcaskill reviewed (changes requested):
The scope of this pull request should also include any other references to `locomotivemtl/charcoal-*` in other files of this repository: README, Travis CI, etc.

pascalrioux committed:
Remove references to locomotivemtl/charcoal-*

mcaskill commented on README.md:
```diff
@@ -329,22 +329,22 @@ Charcoal is licensed under the MIT license. See [LICENSE](LICENSE) for details.
+[core]:             https://packagist.org/packages/charcoal/core
+[factory]:          https://packagist.org/packages/charcoal/factory
+[object]:           https://packagist.org/packages/charcoal/object
+[translator]:       https://packagist.org/packages/charcoal/translator
```
By stripping `charcoal-*`, you are breaking links referencing these URIs.
I suggest changing them to `charcoal/*` to match package names. This way you can use Markdown's shorthand for linking things. For example:

```md
A [Charcoal][charcoal/charcoal] service for generating a sitemap.
// …
-   **[charcoal/app]**: ^0.8

[charcoal/app]:      https://packagist.org/packages/charcoal/app
[charcoal/charcoal]: https://packagist.org/packages/charcoal/charcoal
```

mcaskill commented on README.md:
```diff
@@ -44,11 +44,11 @@ $ composer require locomotivemtl/charcoal-contrib-sitemap
+-   **[charcoal/core][charcoal-core]**: ^0.3
+-   **[charcoal/factory][charcoal-factory]**: ^0.4
+-   **[charcoal/object][charcoal-object]**: ^0.4
+-   **[charcoal/translator][charcoal-translator]**: ^0.3
```
Version numbers should reflect those of its `composer.json` file.

mcaskill commented on composer.json:
```diff
@@ -22,11 +22,11 @@
-        "locomotivemtl/charcoal-factory": "~0.4",
-        "locomotivemtl/charcoal-object": "~0.4",
-        "locomotivemtl/charcoal-translator": "~0.3"
+        "charcoal/app": "~0.8",
+        "charcoal/core": "~0.3",
+        "charcoal/factory": "~0.4",
+        "charcoal/object": "~0.4",
+        "charcoal/translator": "~0.3"
```
Update version numbers to correspond to the new versioning of [charcoal/charcoal](https://github.com/charcoalphp/charcoal).

Not sure how strict or loose it should be. For example:

* `"charcoal/app": ">=1.0.0"`
* `"charcoal/app": "^4.0.0"`
* `"charcoal/app": ">=1.0.0 <5.0.0"`

/cc @JoelAlphonso

pascalrioux committed:
Update dependencies + Fix links in README

pascalrioux committed:
Update shorthand links

mcaskill commented on README.md:
```diff
@@ -44,11 +44,11 @@ $ composer require charcoal/contrib-sitemap
--   **[charcoal/factory][charcoal-factory]**: ^4.0
--   **[charcoal/object][charcoal-object]**: ^4.0
--   **[charcoal/translator][charcoal-translator]**: ^4.0
+-   **[charcoal/app][charcoal/app]**: ^4.0
+-   **[charcoal/core][charcoal/core]**: ^4.0
+-   **[charcoal/factory][charcoal/factory]**: ^4.0
+-   **[charcoal/object][charcoal/object]**: ^4.0
+-   **[charcoal/translator][charcoal/translator]**: ^4.0
```
This is the markdown link shorthand:

```suggestion
-   **[charcoal/app]**: ^4.0
-   **[charcoal/core]**: ^4.0
-   **[charcoal/factory]**: ^4.0
-   **[charcoal/object]**: ^4.0
-   **[charcoal/translator]**: ^4.0
```

Also, line 10 needs to be updated accordingly:

> ```
> A [Charcoal][charcoal-app] service for generating a sitemap.
> ```

mcaskill committed:
Update Charcoal package references

Changed:
- Cleaned-up Markdown in README.
- Replaced reference to charcoal/app with charcoal/charcoal.
- Added keyword to Composer manifest to improve discoverability.
- Fixed homepage URL in Composer manfiest.
